core/src/com/unciv/ui/worldscreen/WorldScreen.kt

```diff
@@ -604,26 +611,38 @@ class WorldScreen(val gameInfo: GameInfo, val viewingCiv:CivilizationInfo) : Bas
 
     }
 
-    private fun createNewWorldScreen(gameInfo: GameInfo, resize:Boolean=false): WorldScreen {
+    class RestoreState(
+        mapHolder: WorldMapHolder,
+        val selectedCivName: String,
+        val viewingCivName: String,
+        val fogOfWar: Boolean
+    ) {
+        val width = mapHolder.width
+        val height = mapHolder.height
+        val scaleX = mapHolder.scaleX
+        val scaleY = mapHolder.scaleY
+        val scrollX = mapHolder.scrollX
+        val scrollY = mapHolder.scrollY
+    }
+    fun getRestoreState(): RestoreState {
+        return RestoreState(mapHolder, selectedCiv.civName, viewingCiv.civName, fogOfWar)
+    }
 
-        game.gameInfo = gameInfo
-        val newWorldScreen = WorldScreen(gameInfo, gameInfo.getPlayerToViewAs())
+    private fun restore(restoreState: RestoreState) {
 
         // This is not the case if you have a multiplayer game where you play as 2 civs
-        if (!resize && newWorldScreen.viewingCiv.civName == viewingCiv.civName) {
-            newWorldScreen.mapHolder.width = mapHolder.width
-            newWorldScreen.mapHolder.height = mapHolder.height
-            newWorldScreen.mapHolder.scaleX = mapHolder.scaleX
-            newWorldScreen.mapHolder.scaleY = mapHolder.scaleY
-            newWorldScreen.mapHolder.scrollX = mapHolder.scrollX
-            newWorldScreen.mapHolder.scrollY = mapHolder.scrollY
-            newWorldScreen.mapHolder.updateVisualScroll()
+        if (viewingCiv.civName == restoreState.viewingCivName) {
+            mapHolder.width = restoreState.width
+            mapHolder.height = restoreState.height
+            mapHolder.scaleX = restoreState.scaleX
+            mapHolder.scaleY = restoreState.scaleY
+            mapHolder.scrollX = restoreState.scrollX
+            mapHolder.scrollY = restoreState.scrollY
+            mapHolder.updateVisualScroll()
         }
 
-        newWorldScreen.selectedCiv = gameInfo.getCivilization(selectedCiv.civName)
-        newWorldScreen.fogOfWar = fogOfWar
-
-        return newWorldScreen
+        selectedCiv = gameInfo.getCivilization(restoreState.selectedCivName)
+        fogOfWar = restoreState.fogOfWar
```

Comment:
I'm pretty sure these are calculable from the game info, not sure if this convenience is more confusing than it is helpful